richtextfx/src/main/java/org/fxmisc/richtext/EditableStyledDocument.java

```diff
@@ -19,7 +19,12 @@
 
     ReadOnlyStyledDocument<PS, S> snapshot();
 
-    EventStream<PlainTextChange> plainChanges();
+    default EventStream<PlainTextChange> plainChanges() {
+        return richChanges()
+                // only allow changes that added/removed text; exclude style changes
+                .filter(c -> c.insertedLength() != 0 || c.removedLength() != 0)
+                .map(c -> new PlainTextChange(c.position, c.removed.getText(), c.inserted.getText()));
+    }
```

Comment:
Wouldn't it be more efficient to filter it first? Otherwise, we're creating unneeded `PlainTextChange`es:
````java
default EventStream<PlainTextChange> plainChanges() {
    return richChanges()
            // only allow changes that added/removed text; exclude style changes
            .filter(c -> !c.inserted.getText().equals(c.removed.getText()))
            .map(c -> new PlainTextChange(c.position, c.removed.getText(), c.inserted.getText()));
}
````